### ISSUE TITLE
New window created from fullscreen doesn't get focus

### ISSUE THREAD

**Issue by RhvdG** [macvim-dev/macvim#1244]
### Steps to reproduce

1. Run MacVim fullscreen
2. Hit cmd-N to create a new window (this is created as a fullscreen 'space')
3. Type something
4. Notice that what was typed ended up on the first fullscreen window (now not visible), not the new one that should have focus

### Expected behaviour

The newly created window (fullscreen) should get focus.

### Version of Vim and architecture

8.2, x86_64, patches: 1-3455

### Environment

macOS 12.3.1

### How MacVim was installed

Homebrew (I think...)

### Issue has been tested with given configuration

- [X] by running MacVim.app from GUI macOS interface
- [ ] by running vim/gvim/etc installed by MacVim
- [ ] by running other versions of vim (e.g. /usr/bin/vim)

### Issue has been tested with no configuration

- [ ] by running `mvim --clean` (or `gvim`, supplied by MacVim distribution)
- [ ] by running `vim --clean` (in terminal, supplied by MacVim distribution)
- [ ] by running `vim --clean` (in terminal, other suppliers, e.g. /usr/bin/vim)

### Other conditions

- [ ] The both Homebrew packages "vim" and "macvim" are installed

**Comment by dantezhu** [comment]
+1

**Comment by ychin** [comment]
That's weird, I'm not reproducing this. Does it matter if you wait a bit or immediately start typing when the animation is ongoing (it takes like less than a second to finish the animation)? Also, are you on a multiple monitors or single-screen environment? macOS does have some funky behavior with focus when on multi-screen so trying to figure out if that could be related.

**Comment by jtiee** [comment]
I see similar behavior with a multi-monitor setup.

With an existing MacVim window on screen 1, run MacVim via a terminal window on screen 2. The new window opens on screen 2, but focus is given to the MacVim window on screen 1.

**Comment by ychin** [comment]
Hmm actually I think I may have misunderstood what you said. You aren't talking about full-screen window but just regular windows. Let me take a look.

**Comment by jtiee** [comment]
Yes, my note is about the "focus" and not the "full screen" aspect of this issue. Let me know if you'd prefer a separate issue for this.

**Comment by ychin** [comment]
> Yes, my note is about the "focus" and not the "full screen" aspect of this issue. Let me know if you'd prefer a separate issue for this.

Yes I think that's a different issue. Feel free to file it, but I'm not sure if I would do anything on it for now. It's kind of "as designed" depending on how we think about how macOS apps work, but I agree it's a little annoying.
